src/plugins/intel_gpu/src/plugin/compiled_model.cpp

```diff
@@ -137,9 +137,41 @@ std::shared_ptr<ngraph::Function> CompiledModel::GetExecGraphInfo() {
 }
 
 InferenceEngine::Parameter CompiledModel::GetConfig(const std::string &name) const {
+    const bool is_new_api = _plugin->GetCore()->isNewAPI();
     auto it = m_config.key_config_map.find(name);
     if (it != m_config.key_config_map.end()) {
-        return it->second;
+        std::string val = it->second;
+        if (is_new_api) {
+            if (name == ov::enable_profiling) {
+                return val == PluginConfigParams::YES ? true : false;
+            } else if (name == ov::hint::model_priority) {
+                return ov::util::string_to_property(val, ov::hint::model_priority);
+            } else if (name == ov::intel_gpu::hint::host_task_priority) {
+                return ov::util::string_to_property(val, ov::intel_gpu::hint::host_task_priority);
+            } else if (name == ov::intel_gpu::hint::queue_priority) {
+                return ov::util::string_to_property(val, ov::intel_gpu::hint::queue_priority);
+            } else if (name == ov::intel_gpu::hint::queue_throttle) {
+                return ov::util::string_to_property(val, ov::intel_gpu::hint::queue_throttle);
+            } else if (name == ov::intel_gpu::enable_loop_unrolling) {
+                return val == PluginConfigParams::YES ? true : false;
+            } else if (name == ov::cache_dir) {
+                return ov::util::string_to_property(val, ov::cache_dir);
+            } else if (name == ov::hint::performance_mode) {
+                return ov::util::string_to_property(val, ov::hint::performance_mode);
+            } else if (name == ov::compilation_num_threads) {
+                return ov::util::string_to_property(val, ov::compilation_num_threads);
+            } else if (name == ov::streams::num) {
+                return ov::util::string_to_property(val, ov::streams::num);
+            } else if (name == ov::hint::num_requests) {
+                return ov::util::string_to_property(val, ov::hint::num_requests);
+            } else if (name == ov::device::id) {
+                return ov::util::string_to_property(val, ov::device::id);
+            } else {
+                IE_THROW() << "Unsupported GPU Plugin Property : " << name;
```

Comment:
1. let's try to return `val`
2. optional: check that if we remove all explicit `if else`, then it works as expected